### PR TITLE
HS-1097 Alert Service: Monitor Policy data model and gRPC APIs

### PR DESCRIPTION
**Pull request by yangli2014**
## Jira link(s)
- https://issues.opennms.org/browse/HS-1097

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts

### CODE REVIEW

File: alert/src/main/java/org/opennms/horizon/alertservice/db/entity/MonitorPolicy.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+@Table(name = "monitoring_policy")
+@Getter
+@Setter
+public class MonitorPolicy extends TenantAwareEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(name = "policy_name")
+    private String name;
+    private String memo;
```
Comment (james8hut) [inline alert/src/main/java/org/opennms/horizon/alertservice/db/entity/MonitorPolicy.java:60]: Should there be a Column annotation above memo?

File: alert/src/main/java/org/opennms/horizon/alertservice/db/entity/SNMPEvent.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Table(name = "trigger_event")
+@Getter
+@Setter
+public class SNMPEvent extends TenantAwareEntity {
```
Comment (james8hut) [inline alert/src/main/java/org/opennms/horizon/alertservice/db/entity/SNMPEvent.java:53]: Should this class not be called TriggerEvent? Or the table called snmp_event? Seems a little confusing to call them different names.

File: alert/src/main/java/org/opennms/horizon/alertservice/db/entity/SNMPEvent.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    private SNMPEventType triggerEvent;
+    @Column(name = "event_count")
+    private Integer count;
+    @Column(name = "over_time")
+    private Integer overtime;
+    @Enumerated(EnumType.STRING)
+    @Column(name = "over_time_unit")
+    private OverTimeUnit overtimeUnit;
+    @Enumerated(EnumType.STRING)
+    private Severity severity;
```
Comment (james8hut) [inline alert/src/main/java/org/opennms/horizon/alertservice/db/entity/SNMPEvent.java:68]: Does Severity not need a Column annotation as well? I see that there is a severity column in the liquibase schema.

File: alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class MonitorPolicyService {
+    private final MonitorPolicyMapper policyMapper;
+    private final MonitorPolicyRepository repository;
+
+    public MonitorPolicyProto creatPolicy(MonitorPolicyProto request, String tenantId) throws IOException {
```
Comment (james8hut) [inline alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java:53]: typo in method name creat**e**Policy

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java [hunk 5]
```diff
@@ -71,14 +70,7 @@ public class AlertTestSteps {
...
     private AlertGrpcClientUtils clientUtils;
-
-    //
-    // Test Configuration
-    //
-    private String applicationBaseHttpUrl;
-    private String applicationBaseGrpcUrl;
-    private String kafkaBootstrapUrl;
-
+    private BackGroundSteps backGround;
```
Comment (james8hut) [inline alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java:73]: nit: background is a single word, so should not have the G capitalised.

File: alert/src/main/java/org/opennms/horizon/alertservice/db/entity/MonitorPolicy.java (hunk 1 above)
Comment (yangli2014) [reply alert/src/main/java/org/opennms/horizon/alertservice/db/entity/MonitorPolicy.java:60]: It is optional if the field name is exactly the same as the column name in the DB table.

File: alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    private final MonitorPolicyRepository repository;
+
+    public MonitorPolicyProto createPolicy(MonitorPolicyProto request, String tenantId) throws IOException {
+        MonitorPolicy policy = policyMapper.protoToEntity(request);
+        updateData(policy, tenantId);
+        MonitorPolicy newPolicy = repository.save(policy);
+        return policyMapper.entityToProto(newPolicy);
+    }
+
+    @Transactional
```
Comment (thomasbigger584) [inline alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java:60]: these `@Transactional`s are read/write - can use `@Transactional(readOnly = true)` to make them for ready only find methods

File: alert/src/test/java/org/opennms/horizon/alertservice/CucumberRunnerIT.java [hunk 7]
```diff
@@ -46,7 +46,7 @@
     glue = {"org.opennms.horizon.alertservice.stepdefs"},
     plugin = {"json:target/cucumber-report.json", "html:target/cucumber.html", "pretty"},
     objectFactory = io.cucumber.picocontainer.PicoFactory.class,
-    features = "classpath:org/opennms/horizon/alertservice"
+    features = "classpath:org/opennms/horizon/alertservice", tags = "@Dev"
```
Comment (thomasbigger584) [inline alert/src/test/java/org/opennms/horizon/alertservice/CucumberRunnerIT.java:49]: does all the tests here still run in alert service with this tags field ?

**Review by thomasbigger584 — APPROVED** [review]

**Pull request merged**